src/Riok.Mapperly/Descriptors/SymbolAccessor.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Helpers;
+using Riok.Mapperly.Symbols;
+
+namespace Riok.Mapperly.Descriptors;
+
+public class SymbolAccessor
+{
+    private readonly WellKnownTypes _types;
+    private readonly Dictionary<ISymbol, ImmutableArray<AttributeData>> _cachedAttributes = new(SymbolEqualityComparer.Default);
```

Comment:
I'd remove the `_cached` prefix, as the other cache fields don't have it either.